### PR TITLE
Fix rerendering bug

### PR DESCRIPTION
**Pull request by leofang**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
@conda-forge/core As discussed on Gitter (https://gitter.im/conda-forge/conda-forge.github.io?at=6004fff8252c0a6ded18cdd4), the first commit e1a77b0 doesn't seem to be the right solution, but it's the only one I found to unblock rerendering. Thoughts?

File: recipe/conda_build_config.yaml
Comment (beckermr) [inline recipe/conda_build_config.yaml:9]: Why is this section in the recipe at all?

Comment (leofang) [reply recipe/conda_build_config.yaml:9]: I am not 100% sure since I wasn't a maintainer until a few days ago, but I thought this is perfectly legal?
https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=pin_run_as_build#building-mpi-variants

Comment (leofang) [reply recipe/conda_build_config.yaml:9]: The strangest thing to me is that it's totally fine to have `mpich` and `msmpi` here, but not `openmpi`.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/conda_build_config.yaml
Comment (beckermr) [reply recipe/conda_build_config.yaml]: I tried a commit of an alternative syntax. This feels like a conda-build bug maybe?

**Comment by beckermr** [comment]
Well that worked! IDK why @leofang but my guess is that this is a safe fix.

**Comment by beckermr** [comment]
I'd hold off on merging though. The ci support files look odd, with openmpi info in the mpich ones. :/

**Comment by leofang** [comment]
> I'd hold off on merging though. The ci support files look odd, with openmpi info in the mpich ones. :/

Agreed. I am also looking into it locally, but I am not sure if I wanna dive deep into conda-build/conda-smithy internals again for this...😂

File: recipe/conda_build_config.yaml
Comment (leofang) [reply recipe/conda_build_config.yaml]: I'd guess so. With your fix, both `openmpi`'s `max_pin=x.x` and pinned version `4.0` appear in ci support files, but not for other 2 mpi impls. As if somewhere in the toolchain `openmpi` is picked out.

Comment (beckermr) [reply recipe/conda_build_config.yaml]: This could be due to the appearance of only openmpi in the build section of the recipe compared to the other providers.

Comment (leofang) [reply recipe/conda_build_config.yaml]: Ah, it would make sense.

Comment (beckermr) [reply recipe/conda_build_config.yaml]: You could rerender with that line in the recipe removed to see if it makes a difference.

Comment (leofang) [reply recipe/conda_build_config.yaml]: >> This could be due to the appearance of only openmpi in the build section of the recipe compared to the other providers.
>>
> You could rerender with that line in the recipe removed to see if it makes a difference.

Actually this is the correct fix! 🎉 Looks like a bad interaction between Jinja and the selector...Pushing a fix shortly.

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
@beckermr Thanks for working on this with me! I think your idea was right that the appearance of `openmpi` in Jinja somehow caused strange behavior. By switching to the selector the ci support files look normal now. PTAL.

**Review by beckermr — APPROVED** [review]
Well done! LGTM!

**Pull request merged**
